crates/e2e/src/setup/onchain_components.rs

```diff
@@ -261,6 +269,106 @@ impl Deref for CowToken {
     }
 }
 
+/// Wrapper over a deployed Safe.
+pub struct Safe {
+    chain_id: U256,
+    contract: GnosisSafe,
+    owner: TestAccount,
+}
+
+impl Safe {
+    /// Return a wrapper at the deployed address.
+    pub fn deployed(chain_id: U256, contract: GnosisSafe, owner: TestAccount) -> Self {
+        Self {
+            chain_id,
+            contract,
+            owner,
+        }
+    }
+
+    /// Returns the address of the Safe.
+    pub fn address(&self) -> H160 {
+        self.contract.address()
+    }
+
+    /// Returns the domain separator for the Safe.
+    fn domain_separator(&self) -> DomainSeparator {
+        let mut buffer = [0_u8; 96];
+        buffer[0..32].copy_from_slice(&hex!(
+            "47e79534a245952e8b16893a336b85a3d9ea9fa8c573f3d803afb92a79469218"
```

Comment:
What are these magic numbers?

Reply:
Ah sorry - meant to include links to code but forgot. Will update.

Reply:
Added some documentation with links to the code that was ported and where the magic numbers come from.